netty/src/test/java/io/grpc/transport/netty/NettyClientTransportTest.java

```diff
@@ -157,15 +163,61 @@ public void creatingMultipleTlsTransportsShouldSucceed() throws Exception {
     }
   }
 
+  @Test
+  public void bufferedStreamsShouldBeClosedWhenConnectionTerminates() throws Exception {
+    // Create the protocol negotiator.
+    File clientCert = TestUtils.loadCert("ca.pem");
+    SslContext clientContext = GrpcSslContexts.forClient().trustManager(clientCert).build();
+    ProtocolNegotiator negotiator = ProtocolNegotiators.tls(clientContext, address);
+
+    NettyClientTransport transport = new NettyClientTransport(address, NioSocketChannel.class,
+            group, negotiator, DEFAULT_WINDOW_SIZE, DEFAULT_WINDOW_SIZE);
+    transports.add(transport);
+    transport.start(clientTransportListener);
+
+    // Wait for the settings to be propagated to the client so that it will respect
+    // SETTINGS_MAX_CONCURRENT_STREAMS.
+    Thread.sleep(1000);
```

Comment:
Can we do an RPC instead? Once the response comes back, we know that settings has been received.